### DISCUSSION TITLE
Retrieval by primary key/custom key

### DISCUSSION THREAD

**Discussion by yairlenga** [prest/prest#710] [API Server] [answered]
Hi,

I am working on a project to expose rest api from postgresql. One of the requirements is to support select/update/delete by unique keys (PK, or other unique).

for example: get db/schema/USA

I believe prestd currently support get db/scheme/countries?id=usa

is it possible to implement custom queries (and updates, delete) when  parameters will be encoded in the url ? Ideally, prestd will spect all unique indices and allow operations by specifying something like
Get /db/schema/table/idxname/field1/field2/…

Is there a way to implement mapping between path parameters and parameters ? 

Thanks, yair lenga

**Comment by arxdsilva** [comment, marked as answer]
Hi @yairlenga, pRest also has custom queries that you can read about [here](https://docs.prestd.com/prestd/docs/api-reference/queries/#scripts-templates-rules). They enable read/write/delete with golang's templates, so you can pass any parameter and use it in your query.

**Reply by avelino** [reply]
Custom query prestd documentation https://docs.prestd.com/prestd/api-reference/queries/
